src/components/Item/VisualizationItem/Visualization/Visualization.js

```diff
@@ -84,29 +83,21 @@ class Visualization extends React.Component {
                 return (
                     <>
                         {!this.state.pluginLoaded && (
-                            <div style={pluginProps.style}>
-                                <LoadingMask />
-                            </div>
+                            <LoadingMask style={pluginProps.style} />
                         )}
-                        <D2Shim d2Config={{}}>
-                            {({ d2 }) => (
-                                <VisualizationPlugin
-                                    d2={d2}
-                                    visualization={this.memoizedGetFilteredVisualization(
-                                        pluginProps.visualization,
-                                        pluginProps.itemFilters
-                                    )}
-                                    onLoadingComplete={this.onLoadingComplete}
-                                    forDashboard={true}
-                                    style={pluginProps.style}
-                                    userSettings={{
-                                        displayProperty: this.context
-                                            .userSettings
-                                            .keyAnalysisDisplayProperty,
-                                    }}
-                                />
+                        <DataVisualizerPlugin
+                            visualization={this.memoizedGetFilteredVisualization(
+                                pluginProps.visualization,
+                                pluginProps.itemFilters
                             )}
-                        </D2Shim>
+                            onLoadingComplete={this.onLoadingComplete}
+                            forDashboard={true}
+                            style={pluginProps.style}
+                            userSettings={{
+                                displayProperty: this.context.userSettings
+                                    .keyAnalysisDisplayProperty,
+                            }}
```

Comment:
`forDashboard`, `userSettings` and `onLoadingComplete` could be moved to DataVisualizerPlugin. Should be pretty straightforward (onLoadingComplete just sets a state var that is only relevant for DataVisualizerPlugin).  But if you don't have time for this, I can do it. If its not done by noon (either by me or you), then let's just go ahead and merge as is, and I can do this later.

Reply:
Changes committed